### PR TITLE
Add course rating APIs and unit tests

### PR DESCRIPTION
**Pull request by etechdevai**
**Description:**  This PR adds implement for course rating APIs and unit tests

**API Endpoint:** 
http://edx.devstack.lms:18000/api/v1/course_rating/course-v1:edX+DemoX+Demo_Course/

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-3718

**Merge checklist:**

- [ ] All reviewers approved
- [ ] Commits are (reasonably) squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** Please add screenshots for any viewable change.
Screenshot 2021-10-14 at 12 27 31 PM
Screenshot 2021-10-14 at 12 32 00 PM

### CODE REVIEW

File: openedx/features/course_rating/api/v1/serializers.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Serializer for "CourseRating" model
+    """
+    course_rating = serializers.SerializerMethodField()
+
+    class Meta:
+        model = CourseRating
+        fields = ["id", "course", "course_rating", "rating", "comment", "is_approved", "user", "moderated_by"]
+
+    def get_course_rating(self, obj):
+        course_avg_rating = CourseAverageRating.objects.filter(course=obj.course).first()
```
Comment (taimoor-ahmed-1) [inline openedx/features/course_rating/api/v1/serializers.py:20]: shouldn't `Course` and `CourseAverageRating` have one-to-one mapping?

File: openedx/features/course_rating/api/v1/views.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from opaque_keys.edx.keys import CourseKey
+from openedx.features.course_rating.models import CourseRating
+from openedx.features.course_rating.api.v1.serializers import CourseRatingSerializer
+
+
+class CourseRatingViewSet(viewsets.ModelViewSet):
+    """
+    List, Create/Update CourseRating.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
```
[the remark this answers is not in the document]
Comment (asadmanzoor93) [reply openedx/features/course_rating/api/v1/views.py]: Done

File: openedx/features/course_rating/api/v1/serializers.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+class CourseRatingSerializer(serializers.ModelSerializer):
+    """
+    Serializer for "CourseRating" model
+    """
+    course_rating = serializers.SerializerMethodField()
+
+    class Meta:
+        model = CourseRating
+        fields = ["id", "course", "course_rating", "rating", "comment", "is_approved", "user", "moderated_by"]
```
Comment (zubair-arbi) [inline openedx/features/course_rating/api/v1/serializers.py:17]: nit: be consistent with single quote in code and double quotes in docstring and comments.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Two suggestions otherwise changes looks good.

File: openedx/features/course_rating/api/v1/serializers.py (hunk 3 above)
Comment (asadmanzoor93) [reply openedx/features/course_rating/api/v1/serializers.py]: Done

**Pull request merged**